framework/include/cppmicroservices/detail/Threads.h

```diff
@@ -151,7 +151,7 @@ template<class MutexHost>
 class NoWaitCondition
 {};
 
-template<class LockingStrategy = MutexLockingStrategy<>,
+template<class LockingStrategy = MutexLockingStrategy<std::recursive_mutex>,
```

Comment:
this change requires more evaluation. I think this change is needed to support the changes in [framework/src/service/ServiceRegistrationBase.cpp](https://github.com/CppMicroServices/CppMicroServices/pull/728/files#diff-36be34fd05c509bdeed05610be9b89469c8d193ec6ce5431f47f6bfd468e4e72) and I'd like to understand why. Making the locking strategy use `recursive_mutex` is a significant change the locking strategy used within the core framework.